astropy/visualization/wcsaxes/ticks.py

```diff
@@ -0,0 +1,183 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+from matplotlib.lines import Path, Line2D
+from matplotlib.transforms import Affine2D
+from matplotlib import rcParams
+
+
+class Ticks(Line2D):
+    """
+    Ticks are derived from Line2D, and note that ticks themselves
+    are markers. Thus, you should use set_mec, set_mew, etc.
+
+    To change the tick size (length), you need to use
+    set_ticksize. To change the direction of the ticks (ticks are
+    in opposite direction of ticklabels by default), use
+    set_tick_out(False).
+
+    Note that Matplotlib's defaults dictionary :data:`~matplotlib.rcParams`
+    contains default settings (color, size, width) of the form `xtick.*` and
+    `ytick.*`. In a WCS projection, there may not be a clear relationship
+    between axes of the projection and 'x' or 'y' axes. For this reason,
+    we read defaults from `xtick.*`. The following settings affect the
+    default appearance of ticks:
+
+    * `xtick.major.size`
+    * `xtick.major.width`
+    * `xtick.color`
+    """
+
+    def __init__(self, ticksize=None, tick_out=False, **kwargs):
+        if ticksize is None:
+            ticksize = rcParams['xtick.major.size']
+        self.set_ticksize(ticksize)
+        self.set_tick_out(tick_out)
+        # FIXME: tick_out is incompatible with Matplotlib tickdir option
```

Comment:
I think I should be able to fix it, but again it will be easier in a separate PR